src/Raven.Server/Documents/Sharding/Handlers/Admin/ShardedAdminShardingHandler.cs

```diff
@@ -13,5 +27,177 @@ public async Task ExecuteMoveDocuments()
                                                                              "This operation is available only from a specific shard"))
                 await processor.ExecuteAsync();
         }
+
+        [RavenShardedAction("/databases/*/admin/sharding/prefixed/add", "POST")]
+        public async Task AddPrefixedShardingSetting()
+        {
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var json = await context.ReadForMemoryAsync(RequestBodyStream(), GetType().Name);
+                var setting = JsonDeserializationCluster.PrefixedShardingSetting(json);
+
+                var shardingConfiguration = ServerStore.Cluster.ReadShardingConfiguration(DatabaseName);
+                ShardingStore.AssertValidPrefix(setting, shardingConfiguration);
+
+                var exists = shardingConfiguration.Prefixed.BinarySearch(setting, PrefixedSettingComparer.Instance) >= 0;
+                if (exists)
+                    throw new InvalidOperationException(
+                        $"Prefix '{setting.Prefix}' already exists in {nameof(ShardingConfiguration)}.{nameof(ShardingConfiguration.Prefixed)}. please use '{nameof(UpdatePrefixedShardingSettingOperation)} operation'");
+
+                var clusterTopology = ServerStore.GetClusterTopology(context);
+                var urls = shardingConfiguration.Orchestrator.Topology.Members.Select(clusterTopology.GetUrlFromTag).ToArray();
+
+                if (await AssertNoDocumentsStartingWith(context, setting.Prefix, urls) == false)
```

Comment:
this is a remote call under read tx, can we scope it only to the `GetClusterTopology`?

Reply:
fixed